M1FExplained/Chapter25/Exercises/Exercise05.lean

```diff
@@ -0,0 +1,45 @@
+import Mathlib.Algebra.Group.Defs
+import Mathlib.Algebra.GroupPower.Basic
+import Mathlib.Tactic.NthRewrite
+import Mathlib.Data.Nat.Basic
+
+/-
+(a) Let G be a group with the property that (ab)^2 = a^2b^2 for all a, b ∈ G. Prove that G is abelian.
+(b) Let G be a group with the property that (ab)^i = a^ib^i ∀a, b ∈ G for three consecutive values of i.
+  Prove that G is abelian.
+(c) Show that the conclusion of part (b) does not follow if we only assume that (ab)^i = a^ib^i
+  for two consecutive values of i.
+-/
+
+
+
+---part a
+
+example {G : Type u_1} [Group G] : ∀ (a b : G), (a*b)^2 = a^2*b^2 → a*b = a*b := by
+  intro a b h₁
```

Comment:
I don't think you've formalised the statement correctly. It says "(for all a and b, (ab)^2=a^2b^2) implies (for all x and y, xy=yx); there are two "forall"s in the question, and only one in your formalisation of it. This should be a relatively simple change though.